lib/github_changelog_generator/fetcher.rb

```diff
@@ -0,0 +1,209 @@
+require "logger"
+
+module GitHubChangelogGenerator
+  # A Fetcher responsible for all requests to GitHub and all basic manipulation with related data
+  # (such as filtering, validating, e.t.c)
+  #
+  # Example:
+  # fetcher = GitHubChangelogGenerator::Fetcher.new options
+  class Fetcher
+    PER_PAGE_NUMBER = 30
+    GH_RATE_LIMIT_EXCEEDED_MSG = "Warning: GitHub API rate limit (5000 per hour) exceeded, change log may be " \
+        "missing some issues. You can limit the number of issues fetched using the `--max-issues NUM` argument."
+
+    def initialize(options = {})
+      @options = options
+
+      @user = @options[:user]
+      @project = @options[:project]
+      @github_token = fetch_github_token
+      @tag_times_hash = {}
+
+      @logger = Logger.new(STDOUT)
+      @logger.formatter = proc do |severity, datetime, progname, msg|
```

Comment:
Unused block argument - `severity`. If it's necessary, use `_` or `_severity` as an argument name to indicate that it won't be used.<br>Unused block argument - `datetime`. If it's necessary, use `_` or `_datetime` as an argument name to indicate that it won't be used.<br>Unused block argument - `progname`. If it's necessary, use `_` or `_progname` as an argument name to indicate that it won't be used.